ui/src/views/access-point/ap-form-step/create-ap.form-body.tsx

```diff
@@ -0,0 +1,149 @@
+import { useQuery } from '@apollo/client';
+import { ethers } from 'ethers';
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import {
+  Button,
+  CardTag,
+  Flex,
+  Form,
+  NFAIcon,
+  Spinner,
+  Stepper,
+  Text,
+} from '@/components';
+import { getNFADocument } from '@/graphclient';
+import { AppLog } from '@/utils';
+
+import { CreateAccessPoint } from '../create-ap.context';
+import { useAccessPointFormContext } from './create-ap.form.context';
+import { useAccount } from 'wagmi';
+
+export const SelectedNFA: React.FC = () => {
+  const { nfa } = CreateAccessPoint.useContext();
+
+  return (
+    <Flex
+      css={{
+        justifyContent: 'space-between',
+      }}
+    >
+      <Flex css={{ alignItems: 'center', maxWidth: '65%' }}>
+        <NFAIcon image={nfa.logo} color={nfa.color} />
+        <Text
+          css={{
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap',
+            overflow: 'hidden',
+          }}
+        >
+          {nfa.name}
+        </Text>
+      </Flex>
+      <CardTag css={{ minWidth: '$28' }}>Selected NFA</CardTag>
+    </Flex>
+  );
+};
+
+export const CreateAccessPointFormBody: React.FC = () => {
+  const { id } = useParams();
+  const { address } = useAccount();
+  const { nextStep } = Stepper.useContext();
+  const { nfa, setNfa, billing } = CreateAccessPoint.useContext();
+  const { setArgs } = CreateAccessPoint.useTransactionContext();
+
+  const {
+    form: {
+      domain: {
+        value: [domain],
+      },
+      isValid: [isValid],
+    },
+  } = useAccessPointFormContext();
+
+  const {
+    form: { domain: domainContext },
+  } = useAccessPointFormContext();
+
+  const {
+    data: nfaData,
+    error: nfaError,
+    loading: nfaLoading,
+  } = useQuery(getNFADocument, {
+    skip: id === undefined,
+    variables: {
+      id: ethers.utils.hexlify(Number(id)),
+    },
+  });
+
+  useEffect(() => {
+    if (nfaError) {
+      AppLog.errorToast('Error fetching NFA');
+    }
+  }, [nfaError]);
+
+  useEffect(() => {
+    if (nfaData) {
+      if (nfaData.token && id) {
+        const { name, tokenId, logo, color } = nfaData.token;
+        setNfa({ name, tokenId, logo, color });
+      } else {
+        AppLog.errorToast("We couldn't find the NFA you are looking for");
+      }
+    }
+  }, [nfaData, id, setNfa]);

```

Comment:
lets avoid use `useEffect` when it is possible to
```suggestion
  const { loading: nfaLoading } = useQuery(getNFADocument, {
    skip: id === undefined,
    variables: {
      id: ethers.utils.hexlify(Number(id)),
    },
    onCompleted(data) {
      if (data.token && id) {
        const { name, tokenId, logo, color } = data.token;
        setNfa({ name, tokenId, logo, color });
      } else {
        AppLog.errorToast("We couldn't find the NFA you are looking for");
      }
    },
    onError(error) {
      AppLog.errorToast('Error fetching NFA', error);
    },
  });
```